lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -1519,10 +1519,14 @@ func (d *btrfs) BackupVolume(vol VolumeCopy, tarWriter *instancewriter.InstanceT
 
 	if len(snapshots) > 0 {
 		// Check requested snapshot match those in storage.
-		err := vol.SnapshotsMatch(snapshots, op)
+		ok, err := d.HasVolumeSnapshots(vol.Volume, vol.Snapshots, op)
```

Comment:
Shall we call it d.CheckVolumeSnapshots() and then use the error returned from it such that it contains the specific error(s) it has encountered (such as specific missing snapshot) rather than using the bool return var?